# 1-Loading dataset & overview

First step is reading the dataset to get an intial look at its composition


In [2]:
import pandas as pd 
df = pd.read_csv("concertation_logement_etudiants_en_metropole.csv")
df.head(5)
# Display basic information and first few rows

ParserError: Error tokenizing data. C error: Expected 19 fields in line 4, saw 20


Unfortunately we got an error indicating the file has some formatting issues, we will come back to them later during the preprocessing phase, for now we will ignore the problematic lines

In [6]:
df = pd.read_csv("concertation_logement_etudiants_en_metropole.csv", on_bad_lines="skip")
df.head(5)

1_niveau_d_etude              2_tu_etudies 3_es_tu_un_etudiant_etranger  \
0               Master       Au sein d'une école                          Non   
1               Master  Au sein d'une université                          Non   
2  BTS - DUT - Licence  Au sein d'une université                          Non   
3  BTS - DUT - Licence  Au sein d'une université                          Non   
4  BTS - DUT - Licence       Au sein d'une école                          Non   

            4_travailles_tu_si_oui_est_ce 5_ton_logement_est_un  \
0  En dehors de tes études (job étudiant)                    T1   
1                                     NaN                Studio   
2  En dehors de tes études (job étudiant)                 Autre   
3                                     NaN                Studio   
4                                     NaN                    T2   

                          6_ton_logement_est  \
0                             Logement privé   
1                             Logement privé   
2  Dans une résidence universitaire du CROUS   
3                             Logement privé   
4                            Logement social   

                                   7_avec_qui_vis_tu  \
0                                          En couple   
1                                               Seul   
2                                               Seul   
3                                               Seul   
4  Hébergé chez la famille (parents, grands-paren...   

  8_selon_toi_ton_logement_est_il  \
0        Loin de ton lieu d'étude   
1        Loin de ton lieu d'étude   
2        Près de ton lieu d'étude   
3        Loin de ton lieu d'étude   
4        Loin de ton lieu d'étude   

  9_parmi_ces_propositions_quelles_sont_les_priorites_pour_un_log  \
0  Un logement spacieux, Un loyer peu élévé (char...                
1  Un loyer peu élévé (charges comprises), Vivre ...                
2  Un loyer peu élévé (charges comprises), Un log...                
3  Vivre seul, Un logement proche du centre-ville...                
4  Un loyer peu élévé (charges comprises), Un log...                

  10_ton_logement_repond_il_aux_priorites_que_tu_as_citees_preced  \
0                                                Non                
1                                                Oui                
2                                                Oui                
3                                                Oui                
4                                                Oui                

                            11_tu_souhaites_preciser  \
0                                                NaN   
1  J'aurais voulu aussi citer \Un logement spacie...   
2                                                NaN   
3                                                NaN   
4  Oui à coté des transports en commun bus juste ...   

  12_comment_as_tu_trouve_ton_logement_actuel_tu_as_rencontre_des  \
0                   J'ai emménagé chez ma partenaire                
1  Sur Leboncoin. \r\nEtant originaire de Savoie,...                
2  J'ai trouvé mon logement grâce à l'ancienne pl...                
3  Sur leboncoin, ça a été compliqué, voire très ...                
4                                                NaN                

  13_penses_tu_changer_de_logement_etudiant_et_si_oui_quelles_en_  \
0  Oui, pour avoir un logement plus spacieux, mie...                
1  Si je reste réaliste, je ne changerais pas de ...                
2  Oui, par manque d'espace. J'ai conscience que ...                
3  j'aimerais bien mais ma famille est trop loin ...                
4                 Oui pour en avoir un en 3 chambres                

  14_quelles_sont_tes_envies_et_attentes_concernant_le_quartier_d  \
0     sécurité, activité, commerces, bonne ambiance.                
1  Habitant entre Wazemmes et Porte des Postes, j...                
2  Avoir un campus universitaire ressemblant à ce...           

In [7]:
print(df.info())
print(df.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278 entries, 0 to 277
Data columns (total 19 columns):
 #   Column                                                           Non-Null Count  Dtype 
---  ------                                                           --------------  ----- 
 0   1_niveau_d_etude                                                 276 non-null    object
 1   2_tu_etudies                                                     278 non-null    object
 2   3_es_tu_un_etudiant_etranger                                     278 non-null    object
 3   4_travailles_tu_si_oui_est_ce                                    191 non-null    object
 4   5_ton_logement_est_un                                            278 non-null    object
 5   6_ton_logement_est                                               275 non-null    object
 6   7_avec_qui_vis_tu                                                277 non-null    object
 7   8_selon_toi_ton_logement_est_il                      

We observe that the dataset contains 278 lines (excluding the misformatted ones), it has 19 columns corresponding to the students' answers to the survey's questions all of them of type object (strings), we are also able to see the number of non null entries for each attribute as well the number of unique values and the most common and its frequency

Next we will give more meaningful names to the dataset's columns in order to make them more easily readable, interpertable and visualisable

In [8]:
new_col_names = [
    "Niveau_Etudes",  # 1_niveau_d_etude
    "Type_Etablissement",  # 2_tu_etudies
    "Etudiant_Etranger",  # 3_es_tu_un_etudiant_etranger
    "Travail_Et_Type",  # 4_travailles_tu_si_oui_est_ce
    "Configuration_Logement",  # 5_ton_logement_est_un
    "Type_Logement",  # 6_ton_logement_est
    "Cohabitation",  # 7_avec_qui_vis_tu
    "Proximite_lieux_Etudes",  # 8_selon_toi_ton_logement_est_il
    "Priorites_Logement",  # 9_parmi_ces_propositions_quelles_sont_les_priorites_pour_un_log
    "Satisfaction_Logement",  # 10_ton_logement_repond_il_aux_priorites_que_tu_as_citees_preced
    "Precisions_Satisfaction",  # 11_tu_souhaites_preciser
    "Methode_Recherche_Logement_Et_Difficulte",  # 12_comment_as_tu_trouve_ton_logement_actuel_tu_as_rencontre_des
    "Volonte_Changement_Logement_Et_Raison",  # 13_penses_tu_changer_de_logement_etudiant_et_si_oui_quelles_en_
    "Attentes_Quartier",  # 14_quelles_sont_tes_envies_et_attentes_concernant_le_quartier_d
    "Fourchette_Loyer",  # 15_quel_est_ton_loyer_mensuel
    "Ressources_Suffisantes_Apres_Loyer",  # 16_une_fois_ton_loyer_paye_estimes_tu_qu_il_te_reste_assez_d_ar
    "Autres_Types_Logement_Parmis_Choix_Multiple",  # 17_connais_tu_ces_autres_types_de_logement
    "Autres_Types_Logement_Libre",  # 18_en_connais_tu_d_autres
    "Volonte_Vivre_Autre_Type"  # 19_aimerais_tu_vivre_dans_ce_type_de_logement
]

df.columns = new_col_names
df.head(5)


Niveau_Etudes        Type_Etablissement Etudiant_Etranger  \
0               Master       Au sein d'une école               Non   
1               Master  Au sein d'une université               Non   
2  BTS - DUT - Licence  Au sein d'une université               Non   
3  BTS - DUT - Licence  Au sein d'une université               Non   
4  BTS - DUT - Licence       Au sein d'une école               Non   

                          Travail_Et_Type Configuration_Logement  \
0  En dehors de tes études (job étudiant)                     T1   
1                                     NaN                 Studio   
2  En dehors de tes études (job étudiant)                  Autre   
3                                     NaN                 Studio   
4                                     NaN                     T2   

                               Type_Logement  \
0                             Logement privé   
1                             Logement privé   
2  Dans une résidence universitaire du CROUS   
3                             Logement privé   
4                            Logement social   

                                        Cohabitation  \
0                                          En couple   
1                                               Seul   
2                                               Seul   
3                                               Seul   
4  Hébergé chez la famille (parents, grands-paren...   

     Proximite_lieux_Etudes  \
0  Loin de ton lieu d'étude   
1  Loin de ton lieu d'étude   
2  Près de ton lieu d'étude   
3  Loin de ton lieu d'étude   
4  Loin de ton lieu d'étude   

                                  Priorites_Logement Satisfaction_Logement  \
0  Un logement spacieux, Un loyer peu élévé (char...                   Non   
1  Un loyer peu élévé (charges comprises), Vivre ...                   Oui   
2  Un loyer peu élévé (charges comprises), Un log...                   Oui   
3  Vivre seul, Un logement proche du centre-ville...                   Oui   
4  Un loyer peu élévé (charges comprises), Un log...                   Oui   

                             Precisions_Satisfaction  \
0                                                NaN   
1  J'aurais voulu aussi citer \Un logement spacie...   
2                                                NaN   
3                                                NaN   
4  Oui à coté des transports en commun bus juste ...   

            Methode_Recherche_Logement_Et_Difficulte  \
0                   J'ai emménagé chez ma partenaire   
1  Sur Leboncoin. \r\nEtant originaire de Savoie,...   
2  J'ai trouvé mon logement grâce à l'ancienne pl...   
3  Sur leboncoin, ça a été compliqué, voire très ...   
4                                                NaN   

               Volonte_Changement_Logement_Et_Raison  \
0  Oui, pour avoir un logement plus spacieux, mie...   
1  Si je reste réaliste, je ne changerais pas de ...   
2  Oui, par manque d'espace. J'ai conscience que ...   
3  j'aimerais bien mais ma famille est trop loin ...   
4                 Oui pour en avoir un en 3 chambres   

                                   Attentes_Quartier    Fourchette_Loyer  \
0     sécurité, activité, commerces, bonne ambiance.       Plus de 450 €   
1  Habitant entre Wazemmes et Porte des Postes, j...  Entre 250 et 450 €   
2  Avoir un campus universitaire ressemblant à ce...  Entre 250 et 450 €   
3  Mon quartier de prédilection c'est le quartier...  Entre 250 et 450 €   
4                     Équipement sportif et culturel    Entre 0 et 250 €   

  Ressources_Suffisantes_Apres_Loyer  \
0                                Non   
1                                Non   
2                                Oui   
3                                Non   
4                     Je ne sais pas   

         Autres_Types_Logement_Parmis_Choix_Multiple  \
0  La cohabitation intergénérationnelle (contribu...   
1  Les colocations solidaires (installation en  c...   
2                   

Now that the dataset's columns is more readbalen let's find out more about the possible values for each column and its types which can either be categorical (answer to a multiple choice question) or free text (text field)

In [28]:
summary = {}
for column in df.columns:
    unique_values = df[column].dropna().unique()  # Drop NaN values
    unique_count = len(unique_values)
    col_type = "Categorical" if unique_count <= 20 else "Free-text"
    range_of_values = unique_values[:10].tolist() # Display the first 10 unique values
    summary[column] = {"Type": col_type, "Range of Values": range_of_values}

categoricals = [column for column in df.columns if summary[column]["Type"] == "Categorical"]
free_texts = df.columns.difference(categoricals).to_list()
# Create a DataFrame with two columns: Categorical and Free-text
max_len = max(len(categoricals), len(free_texts))  # Ensure both columns have the same length
categoricals += [""] * (max_len - len(categoricals))  # Pad with empty strings if needed
free_texts += [""] * (max_len - len(free_texts))

column_types = pd.DataFrame({"Categorical": categoricals, "Free text": free_texts})

# Display the types table
from IPython.display import display
print("1- Classification of the dataset's columns by type")
display(column_types)

print("2- Details for each column")
for col in summary.keys():
    print(f"Column: {col}")
    print("Type:", {summary[col]["Type"]})
    print(f"Unique Values ({len(summary[col]["Range of Values"])}): {summary[col]["Range of Values"]}")  # 
    print("-" * 50)



1- Classification of the dataset's columns by type


Categorical  \
0                                 Niveau_Etudes   
1                            Type_Etablissement   
2                             Etudiant_Etranger   
3                               Travail_Et_Type   
4                        Configuration_Logement   
5                                 Type_Logement   
6                                  Cohabitation   
7                        Proximite_lieux_Etudes   
8                         Satisfaction_Logement   
9                              Fourchette_Loyer   
10           Ressources_Suffisantes_Apres_Loyer   
11  Autres_Types_Logement_Parmis_Choix_Multiple   

                                   Free text  
0                          Attentes_Quartier  
1                Autres_Types_Logement_Libre  
2   Methode_Recherche_Logement_Et_Difficulte  
3                    Precisions_Satisfaction  
4                         Priorites_Logement  
5      Volonte_Changement_Logement_Et_Raison  
6                   Volonte_Vivre_Autre_Type  
7                                             
8                                             
9                                             
10                                            
11

2- Details for each column
Column: Niveau_Etudes
Type: {'Categorical'}
Unique Values (5): ['Master', 'BTS - DUT - Licence', 'Sentiment de sécurité dans les espaces publics', 'Doctorat', 'Donc mes envies']
--------------------------------------------------
Column: Type_Etablissement
Type: {'Categorical'}
Unique Values (4): ["Au sein d'une école", "Au sein d'une université", ' calme."', ' je dirais un quartier \\"attirant\\" dans le réputation aussi."']
--------------------------------------------------
Column: Etudiant_Etranger
Type: {'Categorical'}
Unique Values (4): ['Non', 'Oui', 'Plus de 450 €', 'Entre 250 et 450 €']
--------------------------------------------------
Column: Travail_Et_Type
Type: {'Categorical'}
Unique Values (4): ['En dehors de tes études (job étudiant)', 'Dans le cadre de tes études (stage, apprentissage)', 'Oui', 'Non']
--------------------------------------------------
Column: Configuration_Logement
Type: {'Categorical'}
Unique Values (5): ['T1', 'Studio', 'Autr

As we can see most of the columns are categorical with the only exceptions being: Priorites_Logement, Precisions_Satisfaction, Methode_Recherche_Logement_Et_Difficulte, Volonte_Changement_Logement_Et_Raison, Attentes_Quartier, Autres_Types_Logement_Libre, Volonte_Vivre_Autre_Type


However, as we can see some values appear where they don't belong. Due to misformatting issues some values such as <'Plus de 450 €', 'Entre 250 et 450 €> appear in the column "etudiant_etranger" even though they clearly belong to the column "fourchette_loyer", this is another issue that has to be addressed during preprocessing